torch/csrc/jit/passes/autocast.cpp

```diff
@@ -81,21 +81,49 @@ c10::optional<AutocastScope> parseAutocast(Value* value) {
 void castTensorInputs(Node* node, Symbol cast_op) {
   const auto graph = node->owningGraph();
 
-  WithInsertPoint insert_point(node);
-
   std::unordered_set<Value*> casted_inputs;
   for (auto input : node->inputs()) {
     if (input->type()->kind() == TensorType::Kind) {
       casted_inputs.insert(input);
     }
   }
 
+  WithInsertPoint insert_point(node);
+
   for (auto input : casted_inputs) {
     const auto new_input = graph->insert(cast_op, {input});
     node->replaceInputWith(input, new_input);
   }
 }
 
+void updateDefaultDtype(Node* node, at::ScalarType dtype) {
+  const auto graph = node->owningGraph();
+
+  const auto& actual_args = node->inputs();
+  const auto& formal_args = node->schema().arguments();
+  TORCH_INTERNAL_ASSERT(actual_args.size() == formal_args.size());
+
+  // Try to identify the `dtype` optional paramater
+  Value* dtype_arg = nullptr;
+  for (size_t i = 0; i < formal_args.size(); ++i) {
+    const auto& formal = formal_args[i];
+    if (auto type = formal.type()->cast<OptionalType>()) {
+      if (formal.name() == "dtype" &&
+          type->getElementType()->kind() == TypeKind::IntType) {
```

Comment:
What does it mean that kind() must be an "IntType" here? are the various dtypes represented by IntTypes somehow?

Reply:
Yep - TorchScript maps all integer values to IntType